Distraction Shield/classes/BlockedSite.js

```diff
@@ -0,0 +1,55 @@
+
+function BlockedSite(url, title) {
+    this.constructUrl = function (url) {
+        return "*://" + url + "/*";
+    };
+
+    this.constructIcon = function (url) {
+        return "<img style=\"-webkit-user-select: none\" src=\""+ FAVICONLINK + url +"\">"
+    };
+
+    this.url = this.constructUrl(url);
+    this.domain = url;
+    this.name = title;
+    this.icon = this.constructIcon(url);
+    this.checkboxVal = true;
+    this.counter = 0;
+
+    this.getUrl = function () { return this.url; };
+    this.getDomain = function () { return this.domain; };
+    this.getIcon = function() {return this.icon;};
+    this.getName = function() {return this.name;};
+    this.setCounter = function(newVal) {this.counter = newVal;};
+    this.getCounter = function() {return this.counter;};
+    this.setCheckboxVal = function(newVal) {this.checkboxVal = newVal;};
+    this.getCheckboxVal = function() {return this.checkboxVal;};
+}
+
+/* --------------- --------------- Serialization --------------- --------------- */
+
+//Private to this and storage.js
```

Comment:
well these are functions that can not be defined inside the class since writing to storage makes objects lose all properties, including their functions. Therefore we need to define them outside the class definition. We only want to see these functions here, with their class definitions and the only class that should make actual use of the is the storage. This ensures that we can always just pass an object to the storage and retrieve one from the storage, without having to think about whether we need to serialize or deserialize. It doesn't enforce anything it just the convention we made up, so we put this comment here